### PR TITLE
Issue 16: more slurm versions

### LINKED ISSUE #16: Update to SLURM and Centos stable version
Hi!

Will be nice have Centos 8 and slurm 20.11

### PR DESCRIPTION
**Pull request by LourensVeen**
Hi, and thanks for making and sharing this handy set-up. It's been very useful!

I've been doing some work on https://github.com/vecma-project/QCG-PilotJob, which uses an old fork of slurm-docker-cluster for its test setup. I wanted to test a broad range of Slurm versions, so I made some upgrades, and I figured I'd contribute them back here as well.

This PR upgrades to Rocky Linux 8, which seems to be the natural successor to the old CentOS. (I'm a Debian guy, so I hope I picked the right one.) It also fixes some issues I ran into while compiling different versions of Slurm. For me, this now works with the latest versions of 17.02, 17.11, 18.08, 19.05, 20.02, 20.11 and 21.08.

The final commit makes the Slurm version configurable in docker-compose.yaml, but that does mean that you always have to set it, so it **may break things** for people who have scripts calling docker-compose. You may want to cherry-pick the others rather than merging this wholesale.

This fixes #16 I think, CentOS 8 no longer being available.

### CODE REVIEW

**Comment by giovtorres** [comment]
Hi @LourensVeen, thank you for this contribution!

I was a big fan of CentOS for many years, and I'm still sad that we have to make this change. I'm ok with Rocky Linux, seems like the next logical succession for CentOS.

I like the support for the additional versions. We could use defaults in the docker-compose so that it does not fail for folks.

Something like this where you can provide defaults:

```yaml
slurmdbd:
    image: slurm-docker-cluster:${SLURM_TAG:-slurm-21-08-6-1}
    build:
      context: .
      args: 
        SLURM_TAG: ${SLURM_TAG:-slurm-21-08-6-1:-slurm-slurm-21-08-6-1}
```

Then, you would `docker-compose up` to build and run the default version above, slurm-21-08-6-1, or you can override as follows: `SLURM_TAG=slurm-21-08-6-1 docker-compose build && docker-compose up`.

Thoughts?

Also, another tip to speed up the git clone, you can checkout just the branch you need instead of all since Slurm has git index is pretty big, e.g. 

```shell
git clone -b ${SLURM_TAG} --single-branch --depth=1 https://github.com/SchedMD/slurm.git \
```

instead of this: https://github.com/giovtorres/slurm-docker-cluster/blob/3e6edfd8becbc97570eaee1484540cd2af2e56af/Dockerfile#L55

**Comment by LourensVeen** [comment]
Oh neat, I didn't know docker-compose variables could have default values. Yes, that would be much better.

Maybe there should be two variables, one for the git commit/branch/tag to build and the other one the version tag for the generated image? So you can build git branch `slurm-21.08` and call it`21.08-latest`?

Building takes a lot longer than downloading for me so I hadn't considered checking out only the branch, but it's a good idea to save some bandwidth. All those RPMs are bad enough :-).

File: slurm.conf [hunk 1]
```diff
@@ -23,7 +23,7 @@ SlurmctldPidFile=/var/run/slurmd/slurmctld.pid
 SlurmdPidFile=/var/run/slurmd/slurmd.pid
 ProctrackType=proctrack/linuxproc
 #PluginDir=
-CacheGroups=0
+#CacheGroups=0
```
Comment (giovtorres) [inline slurm.conf:26]: Why the changes to slurm.conf and slurmdbd.conf?

Comment (LourensVeen) [reply slurm.conf:26]: CacheGroups is gone from newer versions of Slurm, so they refuse to start up if you specify it. Since 0 is the default value, I just commented it out. Same thing for the other changes.

**Comment by salanki** [comment]
Chance to get this merged?
